### PR TITLE
Ticket 0022

### PR DESCRIPTION
**Pull request by nfaggian**
This is a test of the pull-request mechanism in github.

### CODE REVIEW

File: register/metrics/metric.py [hunk 1]
```diff
@@ -50,6 +50,8 @@ class Metric(object):
             A particular deformation model.
         warpedImage: nd-array
             Input image after warping.
+        p : list
```
Comment (nfaggian) [inline register/metrics/metric.py]: optional list.

File: register/models/model.py [hunk 2]
```diff
@@ -310,6 +310,128 @@ class Affine(Model):
...
+        coords: nd-array
+           Deformation coordinates.
+        """
+        
+        T = np.array([
+                      [p[0]+1.0, p[2],     p[4]],
+                      [p[1],     p[3]+1.0, p[5]],
+                      [p[6],     p[7],     p[8]+1.0]
+                      ])
+
```
Comment (stefanv) [inline register/models/model.py:404]: If we choose the parameter orders differently, we'll be able to do:

```
T = np.array(p).reshape((3,3))
T += np.eye(3)
```

Comment (riaanvddool) [inline register/models/model.py:404]: My choice was based on the choice for Affine model. So we will need to review both then.

File: register/models/model.py [hunk 3]
```diff
@@ -310,6 +310,128 @@ class Affine(Model):
...
+        dx[:,7] = y * (p[0]*x + p[2]*y + p[4] + x) / (p[6]*x + p[7]*y + p[8] + 1)**2
+        dx[:,8] = 1.0 * (p[0]*x + p[2]*y + p[4] + x) / (p[6]*x + p[7]*y + p[8] + 1)**2
+
+        dy[:,1] = x / (p[6]*x + p[7]*y + p[8] + 1) 
+        dy[:,3] = y / (p[6]*x + p[7]*y + p[8] + 1) 
+        dy[:,5] = 1.0 / (p[6]*x + p[7]*y + p[8] + 1) 
+        dy[:,6] = x * (p[1]*x + p[3]*y + p[5] + y) / (p[6]*x + p[7]*y + p[8] + 1)**2
+        dy[:,7] = y * (p[1]*x + p[3]*y + p[5] + y) / (p[6]*x + p[7]*y + p[8] + 1)**2
+        dy[:,8] = 1.0 * (p[1]*x + p[3]*y + p[5] + y) / (p[6]*x + p[7]*y + p[8] + 1)**2
+
```
Comment (riaanvddool) [inline register/models/model.py:432]: Would be great if someone could check my Jacobian solution here.

Comment (stefanv) [inline register/models/model.py:432]: I found a friend who's willing to do the job:

http://sympy.org

:)

Comment (riaanvddool) [inline register/models/model.py:432]: The friend who helped me is named Wolfram Alpha :)

R

2011/9/21 Stefan van der Walt
reply@reply.github.com:

Comment (stefanv) [inline register/models/model.py:432]: Well played :)

Comment (nfaggian) [inline register/models/model.py:432]: Would be interesting to see sympy scripts for these derivations - might be nice if we ever wanted to latex the derivative computations.

**Pull request merged**